google/cloud/pubsub/internal/tracing_message_batch.cc

```diff
@@ -40,8 +40,40 @@ void TracingMessageBatch::SaveMessage(pubsub::Message m) {
   child_->SaveMessage(std::move(m));
 }
 
-// TODO(#12528): Implement functionality for Flush.
-void TracingMessageBatch::Flush() { child_->Flush(); }
+void TracingMessageBatch::Flush() {
+  using opentelemetry::trace::SpanContext;
+  using AttributesList =
+      std::vector<std::pair<opentelemetry::nostd::string_view,
+                            opentelemetry::common::AttributeValue>>;
+  int64_t message_count = message_spans_.size();
+  auto constexpr kMaxOtelLinks = 128;
+  std::vector<std::pair<SpanContext, AttributesList>> links;
+
+  // If the batch size is less than the max size, add the links to a single
+  // span.
+  if (message_count < kMaxOtelLinks) {
```

Comment:
If you adopt the for loop then `message_count` is not that useful, and we can simply write:

```suggestion
  if (message_spans_.size() < kMaxOtelLinks) {
```